### DISCUSSION TITLE
How to reference multiple models in a mongoose schema referenced path automatically

### DISCUSSION THREAD

**Discussion by samislam** [Automattic/mongoose#12095] [Q&A]
I have the following **schema**:

```js
const transactionSchema =  new mongoose.Schema({
   buyer: {
      type: mongoose.Schema.ObjectId,
      ref: '' // 👈 here
   }
})
```
and I have two **models**:
- VIPClient.
- NormalClient.

I want the VIPClient and the NormalClient to be referenced in the *buyer* **ref** path of the *transactionSchema*.

How?

Is there something in mongoose such as:

```js
const transactionSchema =  new mongoose.Schema({
   buyer: {
      type: mongoose.Schema.ObjectId,
      ref: 'auto' // 👈 here
      // or for example, ref: (model) => model.name,
   }
})
```

??

**Is there a method to make it automatically understand which model was used to create its doc?**

> for a reference here, the importance of the **ref** property is to reference a mongoose model by its name because when you populate this field, it will use the model referenced here. that's why I must add the correct model name, and not any model (💡**hint:** there are middlewares associated with each model that differs from the others 😉).

**Comment by vkarpov15** [comment]
```javascript
const transactionSchema =  new mongoose.Schema({
   buyer: {
      type: mongoose.Schema.ObjectId,
      ref: doc => doc.buyer
   },
   buyer: {
     type: String,
     enum: ['VIPClient', 'NormalClient']
   }
})
```
